esmvalcore/_recipe.py

```diff
@@ -1061,6 +1062,12 @@ def expand_ensemble(variable):
 
         return expanded
 
+    def _add_project_variable_mappings(self, variable):
+        mappings = get_variable_mappings(variable["project"],
+                                         variable["dataset"])
+        mapping = mappings[variable["mip"]][variable["short_name"]]
```

Comment:
it would be a bit cleaner to provide the `mip` and `short_name` as arguments to the `get_variable_mappings` function, to avoid having this code here.